web/controllers/api/docs_controller.ex

```diff
@@ -163,16 +163,13 @@ defmodule HexWeb.API.DocsController do
     HexWeb.CDN.purge_key(:fastly_hexdocs, unversioned_key)
     HexWeb.CDN.purge_key(:fastly_hexdocs, versioned_key)
 
-    # Set docs flag on release
-    HexWeb.Repo.transaction(fn ->
-      Ecto.Changeset.change(release, has_docs: true)
-      |> HexWeb.Repo.update!
+    multi =
+      Ecto.Multi.new
+      |> Ecto.Multi.update(:release, Ecto.Changeset.change(release, has_docs: true))
+      |> Ecto.Multi.update(:package, Ecto.Changeset.change(release.package, docs_updated_at: Ecto.DateTime.utc))
+      |> Ecto.Multi.run(:log, fn _ -> audit(user, "docs.publish", {package, release}) end)
```

Comment:
> My idea about using logger_id was because Ecto would complain if such field was not defined.

Ecto didn't complain unless I tried to do e.g.: `belongs_to :logger, nil` in AuditLog schema - then it complained that the column is missing. When I added belongs_to and the virtual field, it complained that the field is defined twice. I thought I might need the `belongs_to` to get that `logger_id` automagically filled in - I need the id of the audited record in the audit log. On master, as well as on this PR, I'm manually setting the id of the audited record in `audit_log.params` after the first record is created within the same transaction.

> I guess we should support whatever if we decide to add this feature. Would you please send a PR?

Sure! I have a workaround for this PR and I can definitely put this on my TODO list to improve things in Ecto. An upstream PR would be good to discuss all edge cases you had in mind as well.